composer/trainer/trainer.py

```diff
@@ -2911,7 +2912,15 @@ def _eval_loop(
                                 if isinstance(self.state.device, DeviceMPS):
                                     # torchmetrics math has numerical errors on M1 devices
                                     # running the compute on CPU instead
-                                    outputs = self.state.outputs.cpu()
+                                    if isinstance(self.state.outputs, Mapping):
```

Comment:
do we need to account for nesting? I know we've had this discussion wrt torchmetrics before....

Reply:
Hm... I took a quick look and I don't see any issues. If it nests, we just won't move it and there's risk that it gives numerical issues until torchmetrics fixes it which I think is fine